src/MarginModels/PointSetMarginModel.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2020, Battelle Energy Alliance, LLC
+# ALL RIGHTS RESERVED
+"""
+Created on May 13 2021
+
+@author: mandd,wangc
+"""
+
+#External Modules------------------------------------------------------------------------------------
+import abc
+import sys
+import os
+import numpy as np
+import pandas as pd
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from utils import mathUtils as utils
+from utils import InputData, InputTypes
+from .MarginBase import MarginBase
+from scipy.stats import wasserstein_distance
+from sklearn.metrics import pairwise_distances
+#Internal Modules End--------------------------------------------------------------------------------
+
+
+class PointSetMarginModel(MarginBase):
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Collects input specifications for this class.
+      @ In, cls, class instance
+      @ Out, inputSpecs, InputData, specs
+    """
+    inputSpecs = super(PointSetMarginModel, cls).getInputSpecification()
+    inputSpecs.description = """ PointSet Margin Model """
+    inputSpecs.addSub(InputData.parameterInputFactory('failedDataFile',    contentType=InputTypes.InterpretedListType, descr='failed data file'))
+
+
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    MarginBase.__init__(self)
+
+    self.failedDataFileID = None
+    self.columnID = None
+    self.actualDataID = None
+
+
+  def _localHandleInput(self, paramInput):
+    """
```

Comment:
As I suggested before, we can use inheritance strategy and avoid localHandleInput, this is more base class problem.

Reply:
Issue resolved